Assets/MixedRealityToolkit-Tests/Core/TestFixture_01_MixedRealityToolkitTests.cs

```diff
@@ -0,0 +1,543 @@
+            // Tests
+            Assert.IsTrue(success);
+            Assert.IsFalse(isServiceRegistered);
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.IsEmpty(MixedRealityToolkit.RegisteredMixedRealityServices);
+            LogAssert.Expect(LogType.Error, $"Unable to find {typeof(ITestDataProvider1).Name} service.");
+        }
+
+        [Test]
+        public void Test_04_02_02_UnregisterMixedRealityDataProviderByTypeAndName()
+        {
+            // Redo register test service for a valid instance.
+            Test_04_01_RegisterMixedRealityDataProvider();
+
+            // Retrieve service
+            var dataProvider = MixedRealityToolkit.Instance.GetService(typeof(ITestDataProvider1));
+
+            // Validate
+            Assert.IsNotNull(dataProvider);
+
+            var success = MixedRealityToolkit.UnregisterService(typeof(ITestDataProvider1), dataProvider.Name);
+
+            // Validate non-existent service
+            var isServiceRegistered = MixedRealityToolkit.Instance.IsServiceRegistered<ITestDataProvider1>();
+
+            // Tests
+            Assert.IsTrue(success);
+            Assert.IsFalse(isServiceRegistered);
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.IsEmpty(MixedRealityToolkit.RegisteredMixedRealityServices);
+            LogAssert.Expect(LogType.Error, $"Unable to find {typeof(ITestDataProvider1).Name} service.");
+        }
+
+        [Test]
+        public void Test_04_03_RegisterMixedRealityDataProviders()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            // Add test ExtensionService
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider1), new TestDataProvider1("Test Data Provider 1", 10));
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider2), new TestDataProvider2("Test Data Provider 2", 10));
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestFailService), new TestFailService());
+            LogAssert.Expect(LogType.Error, $"{typeof(ITestFailService).Name} does not implement {typeof(IMixedRealityService).Name}.");
+
+            // Retrieve all registered IMixedRealityExtensionServices
+            var extensionServices = MixedRealityToolkit.Instance.GetActiveServices(typeof(IMixedRealityDataProvider));
+
+            // Tests
+            Assert.IsNotNull(MixedRealityToolkit.Instance.ActiveProfile);
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.AreEqual(2, MixedRealityToolkit.RegisteredMixedRealityServices.Count);
+            Assert.AreEqual(extensionServices.Count, MixedRealityToolkit.RegisteredMixedRealityServices.Count);
+        }
+
+        [Test]
+        public void Test_04_04_UnregisterMixedRealityDataProvidersByType()
+        {
+            // Redo register test services for a valid instances.
+            Test_04_03_RegisterMixedRealityDataProviders();
```

Comment:
Rather than re-running a test (which may fail) please consider a helper function that sets the desired initial conditions?

Reply:
Agreed, Tests should be self contained.  If needed, use shared initialisers to "setup" the test, but tests should never be dependant on each other.